### DISCUSSION TITLE
Allow Webservice stock_movements to POST (API) a new stock movement

### DISCUSSION THREAD

**Discussion by jeanbousquet** [PrestaShop/PrestaShop#33395] [Ideas]
As said by Xavier Gouley on 24/Jun/2015, (http://forge.prestashop.com/browse/PSCSX-5988) it would be valuable to allow POST on stock_movements :

" Webservice /api/stock_movements/ allows only GET for instance.

- To have the possibility to modify a stock, that comes from ticket : http://forge.prestashop.com/browse/PSCSX-3170

- and because it's not possible to modify directly the stock table (that has an history through stock_movement table),

If we allow to POST on /api/stock_movements/ to create new stock movements, we'll be able to modify (indirectly) the warehouse stock.
Because a movement can be positive or negative, we'll be able to increase AND decrease the stock amounts."

This would be a clean method for stocks managing with Webservices in PrestaShop because this method keeps the history.

Has this project been taken into account with PrestaShop version 1.7.6 ?
If not, is there an alternative to manage stocks of the products with Webservices ?

Thanks for your help

**Comment by khouloudbelguith** [comment]
Hi @jeanbousquet,

Yes!
You are right, `Method POST is not allowed for the resource stock_movements` with PS1.7.6 also.
Ping @PrestaShop/prestashop-core-developers what do you think?

Thanks!

**Comment by matks** [comment]
Question for @jolelievre : would it be OK to be able to use HTTP POST request on `/api/stock_movements/` to create new stock movements to modify warehouse stock? I know recently you fixed bugs related to this. Or would that put the database in an inconsistent state?
